### PR TITLE
[#127] Fix 'x' behavior at EOL 

### PR DESCRIPTION
**Pull request by tma-isbx**
Determine if we're at the EOL before we modify the text.

Also resolve() when we handle keys in Normal mode (probably a bug). This allows us to test keys.

### CODE REVIEW

File: src/mode/modeNormal.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default class NormalMode extends Mode {
 
             if (keyHandled) {
                 this.keyHistory = [];
-                return this.keyHandler[keysPressed](this.Motion);
```
Comment (jpoon) [inline src/mode/modeNormal.ts]: I think this goes away with the async/await stuff that was just merged a couple of days ago.

**Comment by jpoon** [comment]
Merge conflicts will need to be fixed before I can merge the PR

**Comment by tma-isbx** [comment]
Resolved merge conflicts.

**Comment by jpoon** [comment]
Fixes https://github.com/VSCodeVim/Vim/issues/127

**Pull request merged**